packages/gestalt/src/Masonry/defaultTwoColumnModuleLayout.js

```diff
@@ -40,15 +40,29 @@ function calculateTwoColumnModuleWidth(columnWidth: number, gutter: number): num
   return columnWidth * 2 + gutter;
 }
 
-function calculateSplitIndex(itemsWithoutPositionsLength: number, twoColumnIndex: number): number {
+function calculateSplitIndex(
+  oneColumnItemsLength: number,
+  twoColumnIndex: number,
+  isFirstRow: ?boolean,
+  fitsFirstRow: ?boolean,
+): number {
+  // We add one more item to pre so it is positioned instead of two column item
+  if (isFirstRow && !fitsFirstRow) {
```

Comment:
i don't think this is right actually. this will be true even if the twoColumnItem is not eligible to be in the first row

should this be, instead, something like
```
if (isFirstRow && twoColumnIndex < columnCount && !fitsFirstRow)
```

tested locally and that seems to result in the tests passing